back/engines/free/volunteering/app/services/volunteering/xlsx_service.rb

```diff
@@ -18,11 +18,18 @@ def generate_xlsx pc, volunteers, view_private_attributes: false
       end
       pa = Axlsx::Package.new
       pc.causes.order(:ordering).each do |cause|
-        # Sheet names can only be 31 characters long
-        sheetname = @@multiloc_service.t(cause.title_multiloc)[0..30]
+        sheetname = sanitize_title @@multiloc_service.t(cause.title_multiloc)
         xlsx_service.generate_sheet pa.workbook, sheetname, columns, volunteers.where(cause: cause)
       end
       pa.to_stream
     end
+
+    private
+
+    # Sheet names, derived from Cause titles, can only be 31 characters long, and cannot contain the characters \ , / , * , ? , : , [ , ].
```

Comment:
Moving sanitization of string to this private method doesn't simplify code much, but does also move related comments out of the parent method.

We could also just make this a one-liner in the parent def, and not have the private method.